File: test/services/ts/src/petstore_impl.ts
import { Pet, PetStore } from './generated/petstore/api/index.js';

export class PetStoreImpl {
  private pets: Map<bigint, Pet> = new Map();
  private nextId: bigint = 1n;

  reset(): void {
    this.pets.clear();
    this.nextId = 1n;
  }

  addPet(input: PetStore.methods.addpet.In): PetStore.methods.addpet.Out {
    const id = this.nextId++;
    const pet = new Pet(id, input.name, input.status, input.tag);
    this.pets.set(id, pet);
    return new PetStore.methods.addpet.Out(pet);
  }

  getPet(input: PetStore.methods.getpet.In): PetStore.methods.getpet.Out {
    const pet = this.pets.get(input.id);
    if (pet === undefined) {
      throw new Error(`Pet not found: id=${input.id}`);
    }
    return new PetStore.methods.getpet.Out(pet);
  }

  listPets(_input: PetStore.methods.listpets.In): PetStore.methods.listpets.Out {
    const sorted = [...this.pets.values()].sort((a, b) =>
      a.id < b.id ? -1 : a.id > b.id ? 1 : 0
    );
    return new PetStore.methods.listpets.Out(sorted);
  }

  deletePet(input: PetStore.methods.deletepet.In): PetStore.methods.deletepet.Out {
    const existed = this.pets.delete(input.id);
    return new PetStore.methods.deletepet.Out(existed);
  }
}

File: test/services/py/petstore_impl.py
from Generated.petstore.api.Pet import Pet
from Generated.petstore.api.PetStore import PetStore
from Generated.petstore.api.petstore.addpet.In import In as AddPetIn
from Generated.petstore.api.petstore.addpet.Out import Out as AddPetOut
from Generated.petstore.api.petstore.getpet.In import In as GetPetIn
from Generated.petstore.api.petstore.getpet.Out import Out as GetPetOut
from Generated.petstore.api.petstore.listpets.In import In as ListPetsIn
from Generated.petstore.api.petstore.listpets.Out import Out as ListPetsOut
from Generated.petstore.api.petstore.deletepet.In import In as DeletePetIn
from Generated.petstore.api.petstore.deletepet.Out import Out as DeletePetOut

class PetStoreImpl(PetStore):
    def __init__(self) -> None:
        self._pets: dict[int, Pet] = {}
        self._next_id: int = 1

    def reset(self) -> None:
        self._pets.clear()
        self._next_id = 1

    def addPet(self, arg: AddPetIn) -> AddPetOut:
        pet_id = self._next_id
        self._next_id += 1
        pet = Pet(id=pet_id, name=arg.name, status=arg.status, tag=arg.tag)
        self._pets[pet_id] = pet
        return AddPetOut(pet=pet)

    def getPet(self, arg: GetPetIn) -> GetPetOut:
        pet = self._pets.get(arg.id)
        if pet is None:
            raise KeyError(f"Pet not found: id={arg.id}")
        return GetPetOut(pet=pet)

    def listPets(self, arg: ListPetsIn) -> ListPetsOut:
        sorted_pets = sorted(self._pets.values(), key=lambda p: p.id)
        return ListPetsOut(pets=sorted_pets)

    def deletePet(self, arg: DeletePetIn) -> DeletePetOut:
        existed = arg.id in self._pets
        if existed:
            del self._pets[arg.id]
        return DeletePetOut(deleted=existed)
